runtime/test/plan/plan-producer-test.js

```diff
@@ -66,7 +67,9 @@ class TestPlanProducer extends PlanProducer {
         plan.newSlot('slot0').id = 'id0';
       }
       plan.normalize();
-      plans.push({plan, hash: info.hash});
+      const suggestion = new Suggestion(plan, this.arc);
+      suggestion.hash = info.hash;
```

Comment:
make the hash part of the suggestion constructor, we need to stop doing incidental data pokes like this.

Reply:
added both hash and rank to ctor and made them readonly, thanks.